apis/v1alpha1/tcproute_types.go

```diff
@@ -23,6 +23,11 @@ import (
 type TCPRouteSpec struct {
 	// Rules are a list of TCP matchers and actions.
 	Rules []TCPRouteRule `json:"rules"`
+
+	// Gateways defines which Gateways can use this Route.
```

Comment:
I think this could be interesting but I'd rather wait to make this change until the other larger restructuring gets in since that will directly affect where this field goes.